x/distribution/types/codec.go

```diff
@@ -17,6 +17,8 @@ func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 	cdc.RegisterConcrete(&MsgSetWithdrawAddress{}, "cosmos-sdk/MsgModifyWithdrawAddress", nil)
 	cdc.RegisterConcrete(&MsgFundCommunityPool{}, "cosmos-sdk/MsgFundCommunityPool", nil)
 	cdc.RegisterConcrete(&CommunityPoolSpendProposal{}, "cosmos-sdk/CommunityPoolSpendProposal", nil)
+	cdc.RegisterConcrete(&MsgWithdrawTokenizeShareRecordReward{}, "cosmos-sdk/MsgWithdrawTokenizeShareRecordReward", nil)
+	cdc.RegisterConcrete(&MsgWithdrawAllTokenizeShareRecordReward{}, "cosmos-sdk/MsgWithdrawAllTokenizeShareRecordReward", nil)
```

Comment:
Ledger doesn't support names longer than 39 chars, see https://github.com/cosmos/cosmos-sdk/issues/10870
The newer Cosmos SDK versions use `RegisterAminoMsg()` that does this check and panics if `len() > 39`.

I suggest using `lsm/MsgWithdrawTokenizeShareReward` and `lsm/MsgWithdrawAllTokenizeShareReward`.

Reply:
Yes the prefix chosen to get the msg shorten. Feel free to provide options for those, as we need to agree to anything that is just <= 39 ;)

Reply:
```
cosmos-sdk/MsgWithdrawTokenizeReward
cosmos-sdk/MsgWithdrawAllTokenizeReward
```

?

Reply:
agreed on those!

Reply:
updated [here](https://github.com/cosmos/cosmos-sdk/pull/16747/commits/c65247c6a76d9642d09ddc2761f569fbf4dd8cd3)